web/src/compose.js

```diff
@@ -471,9 +471,17 @@ export function initialize() {
         `.${CSS.escape(compose_banner.CLASSNAMES.wildcard_warning)} .compose_banner_action_button`,
         (event) => {
             event.preventDefault();
-            compose_validate.clear_wildcard_warnings();
-            compose_validate.set_user_acknowledged_wildcard_flag(true);
-            finish();
+            const $edit_form = $(event.target).closest(".edit_form_banners");
+            const $row = $(event.target).closest(".message_row");
+            const $banner_container = $edit_form.length ? $edit_form : $("#compose_banners");
+
+            compose_validate.clear_wildcard_warnings($banner_container);
+            if ($edit_form.length) {
+                message_edit.save_message_row_edit($row);
+            } else {
+                finish();
+            }
+            compose_validate.remove_wildcard_banner_shown_for_container($banner_container);
```

Comment:
Hmm, is this the right time to remove this? I'd expect us to want to do it in the `success` handler for the message actually having been sent / saved when the compose box is cleared if locally echoed.

Reply:
Yeah, the timing wasn't right, so I moved it to those success handlers.